### PR TITLE
Fix logic while parsing the sum statistic for numerical orc columns

### LINKED ISSUE #9182: [BUG] read_orc_statistics incorrectly reads missing statistics values as 0
**Describe the bug**
While reading orc_statistics with cuDF, missing attributes within a column statistics (like sum) are incorrectly read as 0.

**Steps/Code to reproduce bug**
```python
import cudf
import pyorc
from io import BytesIO
import numpy as np

maxint = np.iinfo(np.int64).max

with open("numeric_overflow.orc","wb") as f:
    with pyorc.Writer(f, "struct<a:int,b:int>") as orc_writer:
        orc_writer.write((maxint, 1))
        orc_writer.write((1,1))


file_stats, stripe_stats = cudf.io.orc.read_orc_statistics(["numeric_overflow.orc"])

print(file_stats)
print(stripe_stats)
```

```python
[{'col0': {'number_of_values': 2, 'has_null': False},
  'a': {'number_of_values': 2,
   'has_null': False,
   'minimum': 1,
   'maximum': 9223372036854775807,
   'sum': 0},
  'b': {'number_of_values': 2,
   'has_null': False,
   'minimum': 1,
   'maximum': 1,
   'sum': 2}}]


[{'col0': {'number_of_values': 2, 'has_null': False},
  'a': {'number_of_values': 2,
   'has_null': False,
   'minimum': 1,
   'maximum': 9223372036854775807,
   'sum': 0},
  'b': {'number_of_values': 2,
   'has_null': False,
   'minimum': 1,
   'maximum': 1,
   'sum': 2}}]
```

**Expected behavior**
The statistics values should be read is an `None`

```python
 'a': {'number_of_values': 2,
   'has_null': False,
   'minimum': 1,
   'maximum': 9223372036854775807,
   'sum': None},
```
**Environment overview (please complete the following information)**
 - Environment location: bare-metal
 - Method of cuDF install: conda nightly (21.10)

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

**Additional context**
Seems to be a problem related to ProtoBuf [default](https://developers.google.com/protocol-buffers/docs/proto#optional) values if not checked.

### PR DESCRIPTION
**Pull request by ayushdg**
Fixes #9182.

In cases where the `sum` statistic was not present in the orc file for int and float columns, the values would be incorrectly interpreted as 0 because of protobuf's [default](https://developers.google.com/protocol-buffers/docs/proto#optional) values when fields are missing.

This PR adds a check for field presence before assignment.

### CODE REVIEW

**Comment by ayushdg** [comment]
rerun tests

**Comment by ayushdg** [comment]
Seeing unrelated build errors: 
```
fatal error: An error occurred (404) when calling the HeadObject operation: Key "rapidsai/cudf/pull-request/9183/cpu/flash-cudf-2e3fa15f7bc6e7105d18a7f602302b7fc9994855-11.0-x86_64.tgz" does not exist
```
 rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Review by skirui-source — APPROVED** [review]

**Comment by ayushdg** [comment]
rerun tests

**Comment by ayushdg** [comment]
rerun tests

File: python/cudf/cudf/tests/test_orc.py [hunk 1]
```diff
@@ -1203,3 +1204,73 @@ def test_names_in_struct_dtype_nesting(datadir):
...
+    assert file_stats[0]["a"].get("sum") is None
+    assert file_stats[0]["b"].get("sum") is None
+    assert file_stats[0]["c"].get("sum") == minint64 + 1
+
+    assert stripe_stats[0]["a"].get("sum") is None
+    assert stripe_stats[0]["b"].get("sum") is None
+    assert stripe_stats[0]["c"].get("sum") == minint64 + 1
+
+
+def test_empty_statistics():
```
Comment (rgsl888prabhu) [inline python/cudf/cudf/tests/test_orc.py:1230]: Should there be one more test to check all the statistics value for a proper table of all types of column with values.

**Comment by rgsl888prabhu** [comment]
Rest looks good.

File: python/cudf/cudf/tests/test_orc.py (hunk 1 above)
Comment (ayushdg) [reply python/cudf/cudf/tests/test_orc.py:1230]: Agreed, right now the other read_orc_statistics test only checks for int and bool types. I'll add one/modify existing to include the different column types.

**Review by vuule — APPROVED** [review]
LGTM

**Review by marlenezw — APPROVED** [review]
Looks good on my end!

File: python/cudf/cudf/tests/test_orc.py (hunk 1 above)
Comment (ayushdg) [reply python/cudf/cudf/tests/test_orc.py:1230]: In the interest of time, I'll add one more test in a followup PR.

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**